src/modules/lua/list.c

```diff
@@ -0,0 +1,72 @@
+#include "list.h"
+#include "../../valkeymodule.h"
+
+typedef struct ListNode {
+    void *val;
+    struct ListNode *next;
+} ListNode;
+
+typedef struct List {
+    ListNode *head;
+    int length;
+} List;
+
+typedef struct ListIter {
+    ListNode *current;
+} ListIter;
+
+List *list_create(void) {
+    List *list = ValkeyModule_Calloc(1, sizeof(List));
```

Comment:
Not that it matters much, but calloc doesn't seem to be needed. We could use `ValkeyModule_Alloc`.

```suggestion
    List *list = ValkeyModule_Alloc(sizeof(List));
```